### PR TITLE
packaging: make OSX and Windows packages work

### LINKED ISSUE #241: packaging: problem in OS X packages
Attempted running a GCS binary I'd installed previously and had work. However, after a make clean It dies on splash screen.  It's possible this is spurious / some old weird bin I have sitting around, or it's possible this is more serious.

Running it from terminal yields:

```
Michaels-MBP:MacOS mlyle$ ./dRonin-GCS 
Failed to load core: Could not find 'Core.pluginspec' in /Users/mlyle/dronin/build/ground/gcs/bin/dRonin-GCS.app/Contents/Plugins
Michaels-MBP:MacOS mlyle$ pwd
/Applications/dRonin GCS.app/Contents/MacOS
```

### PR DESCRIPTION
**Pull request by mlyle**
Fixes #241

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/277)

### CODE REVIEW

**Comment by mlyle** [comment]
Works on OSX.  Will test Windows and Linux soon.

**Comment by PTDreamer** [comment]
Why is this needed now? What changed?

**Comment by mlyle** [comment]
@PTDreamer I'm not sure this ever worked.  Before this was putting fully qualified paths into the GCS binary.  Basically, it worked on the original developer's machine as long as you didn't do a make clean afterwards.

I put this back to the old way, which constructs a relative search path and will look at the files in the package.

**Comment by mlyle** [comment]
With these two subsequent changes, the installer on Windows works, except for branding and the welcome screen is blank.

**Comment by PTDreamer** [comment]
> @PTDreamer I'm not sure this ever worked. Before this was putting fully qualified paths into the GCS binary. Basically, it worked on the original developer's machine as long as you didn't do a make clean afterwards.

I see what you mean, that is probably my bad

File: ground/gcs/src/app/main.cpp [hunk 1]
```diff
@@ -140,6 +140,34 @@ static inline QString msgCoreLoadFailure(const QString &why)
...
+    pluginPath += QLatin1Char('/');
+    pluginPath += QLatin1String("Plugins");
+    rc.push_back(pluginPath);
+    return rc;
+}
+
+#ifdef Q_OS_MAC
+#  define SHARE_PATH "/../Resources"
+#else
+#  define SHARE_PATH "/../share/dronin"
```
Comment (PTDreamer) [inline ground/gcs/src/app/main.cpp]: Can you use the external "dronin" string here?

Comment (mlyle) [inline ground/gcs/src/app/main.cpp]: Yes, I should do that.

Comment (mlyle) [inline ground/gcs/src/app/main.cpp]: Actually I have not tested this.. is this supposed to be /../share now?  Could this be why the welcome screen on windows is blank?

**Comment by mlyle** [comment]
@PTDreamer / everyone -- with this commit, startup and welcome screen works on Windows again.

File: ground/gcs/src/app/main.cpp [hunk 2]
```diff
@@ -140,6 +140,34 @@ static inline QString msgCoreLoadFailure(const QString &why)
     return QCoreApplication::translate("Application", "Failed to load core: %1").arg(why);
 }
 
+static inline QStringList getPluginPaths()
```
Comment (mlyle) [inline ground/gcs/src/app/main.cpp]: This is code that was removed in a prior PR for simplified packaging that has instead proven necessary.

File: package/winx86/dronin.nsi [hunk 3]
```diff
@@ -177,6 +177,8 @@ Section "-Resources" InSecResources
   File /r "${GCS_BUILD_TREE}\share\models\*"
   SetOutPath "$INSTDIR\share\pfd"
   File /r "${GCS_BUILD_TREE}\share\pfd\*"
+  SetOutPath "$INSTDIR\share\welcome"
```
Comment (mlyle) [inline package/winx86/dronin.nsi]: The welcome plugin needs to be packaged on Windows for it to work.

File: package/Makefile [hunk 4]
```diff
@@ -74,7 +74,7 @@ package_all: package_ground package_matlab extract_firmware
 
 extract_firmware: $(FLIGHTPKGNAME)
 	mkdir -p "$(PACKAGE_DIR)"/firmware
-	unzip -f "$(FLIGHTPKGNAME)" -d "$(PACKAGE_DIR)"/firmware
+	unzip -u "$(FLIGHTPKGNAME)" -d "$(PACKAGE_DIR)"/firmware
```
Comment (mlyle) [inline package/Makefile]: -f means "freshen" aka create no new files, just update existing ones.  -u "update" is what we want-- overwrite files without fear.

**Pull request merged**